src/conf_mode/protocols_openfabric.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2024 VyOS maintainers and contributors
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 or later as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from sys import exit
+
+from vyos.config import Config
+from vyos.configdict import node_changed
+from vyos.configverify import verify_interface_exists
+from vyos.template import render_to_string
+from vyos import ConfigError
+from vyos import frr
+from vyos import airbag
+
+airbag.enable()
+
+def get_config(config=None):
+    if config:
+        conf = config
+    else:
+        conf = Config()
+
+    base_path = ['protocols', 'openfabric']
+
+    openfabric = conf.get_config_dict(base_path, key_mangling=('-', '_'),
+                                get_first_key=True,
+                                no_tag_node_value_mangle=True)
+
+    # Remove per interface MPLS configuration - get a list if changed
+    # nodes under the interface tagNode
+    interfaces_removed = node_changed(conf, base_path + ['interface'])
+    if interfaces_removed:
+        openfabric['interface_removed'] = list(interfaces_removed)
+
+    if not conf.exists(base_path):
+        openfabric.update({'deleted': ''})
+
+    return openfabric
+
+def verify(openfabric):
+    # bail out early - looks like removal from running config
+    if not openfabric or 'deleted' in openfabric:
+        return None
+
+    if 'net' not in openfabric:
+        raise ConfigError('Network entity is mandatory!')
+
+    # last byte in OpenFabric area address must be 0
+    tmp = openfabric['net'].split('.')
+    if int(tmp[-1]) != 0:
+        raise ConfigError('Last byte of OpenFabric network entity title must always be 0!')
+
+    # If interface not set
+    if 'interface' not in openfabric:
+        raise ConfigError('Interface used for routing updates is mandatory!')
+
+    for iface, iface_config in openfabric['interface'].items():
+        verify_interface_exists(openfabric, iface)
+
+        # If md5 and plaintext-password set at the same time
+        if 'password' in iface_config:
+            if {'md5', 'plaintext_password'} <= set(iface_config['password']):
+                raise ConfigError(f'Can use either md5 or plaintext-password for password for the interface!')
+
+        if iface == 'lo' and 'passive' not in iface_config:
```

Comment:
From a quick test, FRR automatically injects `openfabric passive` when I enable it on `lo`. Making it an error doesn't feel quite right. I think we should add a warning when `passive` is present on `lo` to tell the user that for `lo`, `passive` is implied. 

Or am I missing any factors?

Reply:
@dmbaturin if `passive` mode is not enabled for `lo` in vyos config and I try to modify openfabric configuration and commit changes, then FRR throws an error
```
Failed to execute interface lo  no openfabric passive exit "interface lo
--  no openfabric passive -- exit" we failed to remove this command
Cannot set no passive: loopback is always passive

[[protocols openfabric]] failed
Commit failed
```
That's why I I made it an error

Reply:
I understand that `lo` must be passive. My question is if there are any reasons not to make our template automatically add `openfabric passive` if the interface name is `lo`. Are there any reasons why that's harder than I think it is that I'm missing?

Reply:
Done